include/swift/Reflection/ReflectionContext.h

```diff
@@ -378,29 +379,41 @@ class ReflectionContext
     return readPECOFFSections(ImageStart);
   }
 
-  template <typename T> bool readELFSections(RemoteAddress ImageStart) {
+  template <typename T>
+  bool readELFSections(RemoteAddress ImageStart,
+                       llvm::sys::MemoryBlock FileBuffer) {
+    if (!FileBuffer.base() || !FileBuffer.allocatedSize())
+      return false;
+
     auto Buf =
         this->getReader().readBytes(ImageStart, sizeof(typename T::Header));
```

Comment:
For a later patch: We could get a performance boost if we read everything from the FileBuffer. I assume there's nothing in the image that wouldn't be in the file, too?

Reply:
Yeah, for sure. It wouldn't be much trouble to do this in this patch, actually.